### ISSUE TITLE
Still working?

### ISSUE THREAD

**Issue by Mikan1520** [jwetzell/JakAudioTools#1]
Hey, any chance you're still working on this? I'd love to be able to get all sound out of Jak games

**Comment by jwetzell** [comment]
Not really. Not much more I can do without reversing how the 989 sound driver worked. What sounds are you looking for?

**Comment by Mikan1520** [comment]
I'd love to be able to get them all, but Jak's falling scream from Jak 2 is a big one for me at the moment. I wonder if OpenGOAL could help you with this? I believe they have an implementation of the 989 sound driver in their code

**Comment by Mikan1520** [comment]
Honestly better support for the SBKs in general (such as being able to export from them) would be really nice if you're able. PSound works, but it's difficult to find sounds you need when they're not named.

Also something I noticed, the out_convert.bat doesn't work for the files from the Jak 2 vagwad, not sure if that's a bug or just not really possible. Worked fine for Jak 1 though

**Comment by jwetzell** [comment]
Yeah, I think the file signatures being backwards thwarts ffmpeg being able to automatically detect them. Yeah the SBK is where I sort of put things down because it starts to get into the whole 989 sound driver stuff as the raw sound files are effected by effects (reverb, looping, etc.) to make the actual sound fx you hear.

**Comment by Mikan1520** [comment]
That's fair enough. My hope is that OpenGOAL will support dumping sound files at some point. I've already got every level loaded into Unreal Engine, which is pretty neat haha

**Comment by jwetzell** [comment]
Yeah, that project looks really neat. And looks like they changed SBK file format from 1 to 2 which is why my logic doesn't work. The SBK have a lot of those jak sounds usually in the COMMON SBK files.

**Comment by Mikan1520** [comment]
That makes sense. I was able to find the sounds in Jak 1 pretty easily, but Jak 2 has no falling death scream in common.sbk as far as I could tell from using PSound. But as long as yours supports Jak 1 I suppose I should be fine

**Comment by Mikan1520** [comment]
PSound has an easier time with sample rates on the PS3 version's BNK files over the PS2's SBK, it would seem, so I think I can get whatever I need from there. Still, this really helped me with cutscene audio, thanks for the work you did!

**Comment by jwetzell** [comment]
Yeah, I got into when someone I knew was making some Jak inspired Lo-Fi beats so extracting the samples was really my main goal. I was even able to extract MIDI channels out of the MUS files but still hadn't gotten around to getting the samples/tracks/fx info that is in those MUS files.
![image](https://user-images.githubusercontent.com/18341515/204316311-81a86c82-b50a-4cc9-a324-bce43ea0e204.png)

**Comment by Mikan1520** [comment]
What would need to be done to make sbkparser.py output (preferably named) VAGp files for Jak 1? Would it require an insane amount of changes?

**Comment by jwetzell** [comment]
Not sure, I think a lot of the "parsing" is pretty well done it would just be using that parsed information to chunk up the data and write it out.

**Comment by jwetzell** [comment]
As long as the SBK doesn't use a similar kind of scheme to the MUS files where the sound pieces can be run through some effects like reverb, loop, etc. which some of the output of PSound leads me to believe. But it's been such a long time that I've looked at it's all guessing at this point.

**Comment by Mikan1520** [comment]
Hmm. Alright, thanks

**Comment by Mikan1520** [comment]
After trudging through what I could make of the OpenGOAL code (not my specialty lol) it does seem out of my reach for now. I'll play around with sample rates in PSound I suppose haha
